### ISSUE TITLE
default option values in README

### ISSUE THREAD

**Issue by hb9kns** [gophernicus/gophernicus#41]
The default option values in the README are hardcoded, which currently results e.g in the default charset still being reported as US-ASCII, whilst it is now UTF-8.
The options should be taken from `gophernicus.h` and not hardcoded.
I could add another shell script that runs this from the Makefile...

**Comment by fosslinux** [comment]
I think, that we should also have a seperate README that gets installed to the README.md in the repository. It makes it more accessible to people looking at the github.

**Comment by tallship** [comment]
> 
> 
> Perhaps a dedicated /docs/ here would be a good idea? There's enough that isn't explained in its entirety, such as caps.txt treatment, etc., and other items like longform examples, that could be the basis for a manual.

Scratch that....

Install creates a docs/ with files mentioned in my comment for #45 [here](https://github.com/gophernicus/gophernicus/issues/45#issuecomment-530813652).

At this point, I'm thinking it would be best to have expanded, more complete documentation contained in a Wiki here, instead of polluting the package with a plethora of files that might make getting the most pertinent information from the docs/ difficult to locate.

A reference with a link or links to the Wiki here could be placed in one of the files like README.md or perhaps a single addtional informational file called "first_steps.md" (or .txt) so as to keep the installation and associated docs/ clean.

Just my thoughts upon further reflection from last night.

**Comment by fosslinux** [comment]
Yes, I agree. I dislike the current .md files getting installed; I honestly do not think it is helpful. I'll write a manpage shortly (#9), then we can deprecate the existing documentation and link to the wiki instead here?

So then, while I think of it, do we potentially want to move the website to GitHub Pages? This allows anyone to edit the content of the website via PR and makes it easier for us to add documentation, etc.

@hb9kns @tallship

**Comment by tallship** [comment]
> Yes, I agree. I dislike the current .md files getting installed; I honestly do not think it is helpful. I'll write a manpage shortly (#9), then we can deprecate the existing documentation and link to the wiki instead here?

The files at this time aren't actually installed into /var/gopher/docs/ - they're just symlinked from /user/doc/...

Perhaps for the near future simply removing that link creation in the Makefile would suffice?

Man pages are a great idea and lends itself to Gophernicus being a regular aspect of the system too!

Moving the website would be an ideal situation. The only remaining thing would be maintaining the domain registration after that, and like you said, the website could be maintained with simple pull requests, as well as links to the wiki 👍

**Comment by fosslinux** [comment]
I'm aware they're symlinked. I just don't feel like having these docs is good on a system.

If @hb9kns is cool with it we'll go with that, then!

**Comment by hb9kns** [comment]
I'm not yet sure whether I understood all ideas correctly. In the meantime, I'll note my thoughts here:

- gophernicus and all its documentation should be independent of Github or any other service (actually, this is my belief for *any* source code or original content, which makes git so appealing) -- we just want to make use of Github Pages or Wiki or whatever, but we should keep everything so that it can be moved anywhere if need be; therefore I'm not that fine with wikis, because they make it difficult to move the information elsewhere
- gophernicus.org also just should be a convenient starting point, but not a dependency for gophernicus, so I think it's not better or worse than Github Pages or Wiki -- except that we might mirror content there as well (which is not easily done right now, because I have a cheap hosting plan only allowing for a static and small site)
- manpages are good, because they are where users look for information
- `README.md` should tell people how to get something installed/working, but it does not need to cover all aspects of how it should be used on a system (however, it might be useful from a advertisement viewpoint); and indeed it should not be installed on systems, that's what manpages are for
- documentation for gophernicus should only be in `/docs/` if that makes sense as being accessible through gophernicus, *i.e* like describing gophermaps for users etc; but that should also be in a manpage, or the manpage should point to `/docs/` if we don't want doublettes (actually, the best might be to have all as manpage, and during installation copy the manpage to that folder)

**Comment by fosslinux** [comment]
> gophernicus and all its documentation should be independent of Github or any other service (actually, this is my belief for any source code or original content, which makes git so appealing) -- we just want to make use of Github Pages or Wiki or whatever, but we should keep everything so that it can be moved anywhere if need be; therefore I'm not that fine with wikis, because they make it difficult to move the information elsewhere

I agree, but afaik Github wiki's are only markdown? That can easily be moved to another wiki.

> gophernicus.org also just should be a convenient starting point, but not a dependency for gophernicus, so I think it's not better or worse than Github Pages or Wiki -- except that we might mirror content there as well (which is not easily done right now, because I have a cheap hosting plan only allowing for a static and small site)

Yeah, maybe it's fine as-is. We still need somewhere to put all the information scattered everywhere though, that's why I think we should have a wiki.

> manpages are good, because they are where users look for information
>
> README.md should tell people how to get something installed/working, but it does not need to cover all aspects of how it should be used on a system (however, it might be useful from a advertisement viewpoint); and indeed it should not be installed on systems, that's what manpages are for

I've made a manpage; so #9 is closed. Yay :)

> documentation for gophernicus should only be in /docs/ if that makes sense as being accessible through gophernicus, i.e like describing gophermaps for users etc; but that should also be in a manpage, or the manpage should point to /docs/ if we don't want doublettes (actually, the best might be to have all as manpage, and during installation copy the manpage to that folder)

agreed

**Comment by hb9kns** [comment]
> > gophernicus and all its documentation should be independent of Github or any other service ...

> I agree, but afaik Github wiki's are only markdown? That can easily be moved to another wiki.

In that case, why not, if most people seem to be fine with it? My other concern with wikis is that they need a server to be useful (although it's relatively easy to put up a localhost server if need be, I know), but this may be due to my deep-ingrained and anachronistic aversion to things that cannot directly live as flat files.

> > gophernicus.org also just should be a convenient starting point, but not a dependency ...

> Yeah, maybe it's fine as-is. We still need somewhere to put all the information scattered everywhere though, that's why I think we should have a wiki.

Well, I fully agree that wikis seem to be one of the best ways to foster collaborative documentation!
Therefore, if we think it will speed up assembling of knowledge, then by all means let's go for it!

> > manpages are good, because they are where users look for information ...

> I've made a manpage; so #9 is closed. Yay :)

Great, congrats! ^-^

**Comment by fosslinux** [comment]
Cool, I've made the homepage (just says welcome to the wiki atm). Unfortunatly, there are no issues/PRs on the wiki. The only option is to have just collaborators (me/hb9kns/kim) edit it or allow everyone to edit it freely.

Alternatively, we could make a gophernicus/wiki repo and then I can set up a cronjob on my raspi to pull that and push to the actual wiki hourly.

**Comment by augfab** [comment]
On a related note, I reworked the markup of the manual page there #69.
